compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1973,6 +1974,22 @@ class Typer extends Namer
             val pat1 = if (patType eq patType1) pat else pat.withType(patType1)
             patBuf += pat1
           }
+        case ddef: ValOrDefDef =>
+          if (ddef.symbol.annotations.exists(_.symbol == defn.InternalQuoted_patternBindHoleAnnot)) {
+            val tpe = ddef.symbol.info match {
```

Comment:
Pick a more descriptive name for `tpe`. Maybe `bindingType`?

Reply:
Renamed